data_structures/src/superblock.rs

```diff
@@ -43,31 +43,50 @@ pub enum SuperBlockConsensus {
     Unknown,
 }
 
-/// State related to superblocks
+/// ARS identities
 #[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
-pub struct SuperBlockState {
-    // Set of ARS identities that will be able to send superblock votes in the next superblock epoch
-    current_ars_identities: HashSet<PublicKeyHash>,
-    // current_ars_identities ordered by reputation
-    current_ordered_ars_identities: Vec<PublicKeyHash>,
-    // Subset of ARS in charge of signing the next superblock
-    current_signing_committee: Option<HashSet<PublicKeyHash>>,
-    // Current superblock hash created by this node
-    current_superblock_hash: Hash,
-    // Current superblock index, used to limit the range of broadcasted votes to
-    // [index - 1, index + 1]. So if index is 10, only votes with index 9, 10, 11 will be broadcasted
-    current_superblock_index: u32,
-    // Map of identities that voted more than once. This votes are considered invalid.
-    identities_that_voted_more_than_once: HashMap<PublicKeyHash, Vec<SuperBlockVote>>,
-    // Set of ARS identities that can currently send superblock votes
-    previous_ars_identities: Option<HashSet<PublicKeyHash>>,
-    // previous_ars_identities ordered by reputation
-    previous_ordered_ars_identities: Vec<PublicKeyHash>,
-    // The last ARS ordered keys
-    previous_ars_ordered_keys: Vec<Bn256PublicKey>,
+pub struct ARSIdentities {
+    // HashSet of the identities in a specific ARS
+    identities: HashSet<PublicKeyHash>,
+    // Ordered vector of the identities in a specific ARS
+    ordered_identities: Vec<PublicKeyHash>,
+    // Alternative public Key mapping in a specific ARS
+    alt_keys: AltKeys,
```

Comment:
AltKeys is also present in the ChainState. Since ARSIdentities is part of the superblockState and the latter is also part of the ChainState, shall we remove the field alt_keys from the ChainState?